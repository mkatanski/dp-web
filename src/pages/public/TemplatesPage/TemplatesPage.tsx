import React from "react";
import { PageTitle } from "components/PageTitle";
import { TemplatesTableView } from "views/TemplatesTableView";

const TemplatesPage: React.FC = () => (
  <>
    <PageTitle category="general" title="Templates" />
    <TemplatesTableView />
  </>
);

export default TemplatesPage;
