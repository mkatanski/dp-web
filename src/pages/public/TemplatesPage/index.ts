import TemplatesPage from "./TemplatesPage";
export * from "./TemplatesPage";

export default TemplatesPage;
