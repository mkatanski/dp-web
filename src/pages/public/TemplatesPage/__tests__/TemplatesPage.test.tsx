/* eslint-disable react/display-name */
import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";

import TemplatesPage from "../TemplatesPage";

jest.mock("components/PageTitle", () => ({
  PageTitle: ({ ...rest }: React.PropsWithChildren<{}>) => (
    <div id="PageTitle" data-props={rest} />
  )
}));

jest.mock("views/TemplatesTableView", () => ({
  TemplatesTableView: ({ ...rest }: React.PropsWithChildren<{}>) => (
    <div id="TemplatesTableView" data-props={rest} />
  )
}));

describe("pages", () => {
  describe("protected", () => {
    describe("TemplatesPage", () => {
      it("should render TemplatesPage component", async () => {
        const tree = TestRenderer.create(<TemplatesPage />);
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
