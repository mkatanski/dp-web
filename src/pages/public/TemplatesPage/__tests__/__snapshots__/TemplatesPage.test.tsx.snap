// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages protected TemplatesPage should render TemplatesPage component 1`] = `
Array [
  <div
    data-props={
      Object {
        "category": "general",
        "title": "Templates",
      }
    }
    id="PageTitle"
  />,
  <div
    data-props={Object {}}
    id="TemplatesTableView"
  />,
]
`;
